framework/src/modules/network/network.js

```diff
@@ -47,35 +50,76 @@ module.exports = class Network {
 
 		this.logger = createLoggerComponent(loggerConfig);
 
+		const storageConfig = await this.channel.invoke(
+			'app:getComponentConfig',
+			'storage'
+		);
+		const dbLogger =
+			storageConfig.logFileName &&
+			storageConfig.logFileName === loggerConfig.logFileName
+				? this.logger
+				: createLoggerComponent(
+						Object.assign({}, loggerConfig, {
+							logFileName: storageConfig.logFileName,
+						})
+				  );
+
+		this.storage = createStorageComponent(storageConfig, dbLogger);
+		this.storage.registerEntity('Peer', Peer);
+
+		try {
+			const status = await this.storage.bootstrap();
+			if (!status) {
+				throw new Error('Cannot bootstrap the storage component');
+			}
+		} catch (err) {
+			throw err;
+		}
+
+		// Load peers from the database that were tried or connected the last time node was running
+		let triedPeers;
+		try {
+			triedPeers = await this.storage.entities.Peer.get({}, { limit: null });
+		} catch (err) {
+			throw err;
+		}
 		// TODO: Nonce overwrite should be removed once the Network module has been fully integreated into core and the old peer system has been fully removed.
 		// We need this because the old peer system which runs in parallel will conflict with the new one if they share the same nonce.
 		const moduleNonce = randomstring.generate(16);
 		const sanitizeNodeInfo = nodeInfo => ({
 			...nodeInfo,
 			state: 2, // TODO: Delete state property
 			nonce: moduleNonce,
-			wsPort: this.options.wsPort,
+			wsPort: this.options.default.wsPort,
```

Comment:
I think we should not use `default` as it could be overwritten by custom config.
Same for below

Reply:
But at this point, `this.options.wsPort` is `undefined`

Reply:
They should be combined at some point. The fields from the config file should override the default modules ones. Could be raised as a separate issue if it's not already doing that.

Reply:
It's fixed in PR #3453 , so I have reverted these changes now.